backend/fms_core/services/library.py

```diff
@@ -143,3 +144,71 @@ def convert_library(process, platform, sample_source, container_destination, coo
             errors.append(e)
 
     return library_destination, errors, warnings
+
+def set_library_size(sample, library_size = None):
+    errors = []
+    warnings = []
+
+    if sample is None:
+        errors.append('Missing sample')
+
+    if not sample.is_library:
+        errors.append(f'Sample {sample.name} is not a library.')
+
+    try:
+        library = sample.derived_sample_not_pool.library
+        library.library_size = library_size
+        library.save()
+    except Exception as e:
+        errors.append(str(e))
+
+    return sample, errors, warnings
+
+
+def update_library(sample, **kwargs):
+    errors = []
+    warnings = []
+
+    if sample is None:
+        errors.append('Missing sample')
+    elif not sample.is_library:
+        errors.append('Sample {sample.name} is not a library')
+    else:
+        try:
+            library = sample.derived_sample_not_pool.library
+
+            if 'library_type' in kwargs:
```

Comment:
this could be a very long list of arguments... i would make that more compact by using  the kwargs dict in a model update like in the update sample viewset. You can add some type checking or validation in a compact copy from kwargs to a local dict for the update.